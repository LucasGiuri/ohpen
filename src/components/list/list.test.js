import List from './list';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('List', () => {
  const dataTestPrefix = 'dataTestPrefix';
  const fakeData = [
    'subscription1', 'subscription2', 'subscription3', 'subscription4',
    'subscription100', 'subscription200', 'subscription300', 'subscriptiona4',
    'subscription1000', 'subscription2000', 'subscription0003', 'subscriptions4'
  ];

  describe('Snapshots', () => {
    it('default snapshot render', () => {
      const tree = renderer.create(<List data={fakeData} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with custom dataPrefix snapshot render', () => {
      const tree = renderer.create(<List data={fakeData} dataTestPrefix={dataTestPrefix} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
