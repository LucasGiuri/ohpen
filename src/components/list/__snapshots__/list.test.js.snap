// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshots default snapshot render 1`] = `
<div
  className="sc-bdfBQB jFzuzi"
  data-test="list-list"
>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION1
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION2
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION3
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION4
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION100
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION200
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION300
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTIONA4
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION1000
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION2000
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION0003
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTIONS4
  </div>
</div>
`;

exports[`List Snapshots with custom dataPrefix snapshot render 1`] = `
<div
  className="sc-bdfBQB jFzuzi"
  data-test="dataTestPrefix-list"
>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION1
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION2
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION3
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION4
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION100
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION200
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION300
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTIONA4
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION1000
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION2000
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTION0003
  </div>
  <div
    className="sc-gsTEea kIYsgc"
  >
    SUBSCRIPTIONS4
  </div>
</div>
`;
